providers/shared/components/backupstore/id.ftl

```diff
@@ -0,0 +1,217 @@
+                        "Description" : "Maximum number of days to retain backups",
+                        "Types" : INTEGER_TYPE
+                    },
+
+                ]
+            },
+        ]
+/]
+
+[@addComponentDeployment
+    type=BACKUPSTORE_COMPONENT_TYPE
+    defaultGroup="solution"
+/]
+
+[@addChildComponent
+    type=BACKUPSTORE_RULE_COMPONENT_TYPE
+    parent=BACKUP_COMPONENT_TYPE
+    childAttribute="Rules"
+    linkAttributes="Rule"
+    properties=
+        [
+            {
+                "Type"  : "Description",
+                "Value" : "Defines a schedule of backups and the affected targets"
+            }
+        ]
+    attributes=
+        [
+            {
+                "Names" : "Enabled",
+                "Description" : "Permit the rule to be considered",
+                "Types" : BOOLEAN_TYPE,
+                "Default" : true
+            },
+            {
+                "Names" : "Order",
+                "Description" : "The order in which rules are processed",
+                "Types" : NUMBER_TYPE,
+                "Default" : 100
+            },
+            {
+                "Names" : "Schedule",
+                "Description" : "When to produce snapshots"
+                "Children" : [
+                    {
+                        "Names" : "Expression",
+                        "Description" : "UTC based cron schedule to create backup snapshots"
+                        "Types" : STRING_TYPE
+                    },
+                    {
+                        "Names" : "PointInTimeSupport",
+                        "Description" : "Enable ability for point-in-time restoration"
+                        "Types" : BOOLEAN_TYPE,
+                        "Default" : false
+                    }
+                ]
+            },
+            {
+                "Names" : "StartWindow",
+                "Description" : "Period in minutes in which snapshot must start",
+                "Types" : NUMBER_TYPE
+            },
+            {
+                "Names" : "FinishWindow",
+                "Description" : "Period in minutes after a snapshot has started in which it must finish",
+                "Types" : NUMBER_TYPE
+            },
+            {
+                "Names" : "Lifecycle",
+                "Children" : [
+                    {
+                        "Names" : "Expiration",
+                        "Types" : [NUMBER_TYPE],
+                        "Description" : "Days until snapshot is deleted. Must be greater than Offline (if provided)",
+                        "Default" : "_operations"
+                    },
+                    {
+                        "Names" : "Offline",
+                        "Types" : [NUMBER_TYPE],
+                        "Description" : "Days until backup is moved to longer term storage (if supported)",
+                        "Default" : "_operations"
+                    }
+                ]
+            },
```

Comment:
Types would be STRING_TYPE and NUMBER_TYPE based on the default yeah? 

Also kinda confusing looking at it as the defaults are the same value but it says that the Expiration has to be larger than the Offline

Reply:
I've tweaked the defaults and descriptions. With qualification and modules, I've not tried to replicate the `_` style values we have for S3 - I think they would be better handled by deployment profiles.